### PR TITLE
[#2433] improvement(trino-connector): simplify dynamic catalog names

### PR DESCRIPTION
**Pull request by xiacongling**
### What changes were proposed in this pull request?

Support omitting metalake prefix for Trino connector. 

### Why are the changes needed?

Gravitino register a dynamic catalog with name like `some_metalake.some_catalog`. It is long and must be quoted to use. Besides, if one wants to manage file-based catalogs with Gravitino, users need to adjust their SQL for catalog name changing. With this patch, Trino admins can add a Gravitino connector property `gravitino.simplify-catalog-names=true` to keep the catalog name as it is without `some_metalake.` prefix.

Fix: #2433

### Does this PR introduce _any_ user-facing change?

No for default settings. If `gravitino.simplify-catalog-names=true` is set, the catalog names will change when using Trino with Gravitino.

### How was this patch tested?

UT

### CODE REVIEW

**Comment by xiacongling** [comment]
test cases will be added...

**Comment by diqiu50** [comment]
I think we should consider these factors:
1. If the configuration `gravitino.simplify-catalog-names=true` is set, users can only configure one Gravitino connector.
2. Check if the catalog name conflicts with other connectors.

**Comment by diqiu50** [comment]
Thank for your worksI  
There are two issues that need to be explained.
1、Multiple Gravitino connectors can work in CatalogConnectorManager. All the Gravitino-managed catalog creations need to call GravitinoConnectorFactory#create again. We name these catalogs as dynamic catalogs. The dynamic catalog will be added to CatalogConnectorManager when it is created.

2. Ignore duplicated catalog is ok.  Graviton-managed catalogs are unique, so they might duplicate user-configured catalogs. And that loaded after Trino manager catalogs. .If that happens, we should currently ignore the duplicate catalog and log the error.

**Comment by diqiu50** [comment]
@xiacongling Is your pull request ready now? I think it still lacks some tests and checks for name conflicts.

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoConnectorFactory.java [hunk 1]
```diff
@@ -95,6 +96,13 @@ public Connector create(
       if (Strings.isNullOrEmpty(metalake)) {
         throw new TrinoException(GRAVITINO_METALAKE_NOT_EXISTS, "No gravitino metalake selected");
       }
+      if (config.simplifyCatalogNames()) {
+        if (!catalogConnectorManager.getCatalogs().isEmpty()) {
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoConnectorFactory.java:100]: Merge this two `if` condition.

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java [hunk 2]
```diff
@@ -215,7 +219,11 @@ private void unloadCatalog(GravitinoMetalake metalake, String catalogFullName) {
   }
 
   public CatalogConnectorContext getCatalogConnector(String catalogName) {
-    return catalogConnectors.get(catalogName);
+    CatalogConnectorContext connectorContext = catalogConnectors.get(catalogName);
+    if (connectorContext == null) {
+      connectorContext = catalogConnectors.get(catalogName);
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java:224]: Why do we need to call `catalogConnectors.get(catalogName)` twice?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java [hunk 3]
```diff
@@ -360,6 +368,12 @@ public void alterCatalog(
   }
 
   public void addMetalake(String metalake) {
+    if (config.simplifyCatalogNames()) {
+      if (!usedMetalakes.isEmpty())
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java:372]: What is the size of `usedMetalakes` is 1 and what's the meaning of `Multiple metalakes`?

File: trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnectorWithSimpleCatalogName.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      properties.put("gravitino.simplify-catalog-names", "true");
+      queryRunner.createCatalog("gravitino", "gravitino", properties);
+
+      CatalogConnectorManager catalogConnectorManager =
+          gravitinoPlugin.getCatalogConnectorManager();
+      catalogConnectorManager.setGravitinoClient(gravitinoClient);
+      server.setCatalogConnectorManager(catalogConnectorManager);
+      // Wait for the catalog to be created. Wait for at least 30 seconds.
+      int max_tries = 35;
+      while (catalogConnectorManager.getCatalogs().isEmpty() && max_tries > 0) {
```
Comment (yuqi1129) [inline trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnectorWithSimpleCatalogName.java:56]: We'd better use `awaitability` to replace `Thread.sleep`.

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java (hunk 3 above)
Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java:372]: A `usedMetalakes` value of 1 indicates that the user have configured one Gravitino catalog using a metalake. 
When the user add a new Gravitino catalog and using another metalake,  it''s running on  multiple catalogs mode

**Comment by yuqi1129** [comment]
@diqiu50 
Don't forget to track an issue that makes the default value of this configuration true.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Comment by jerryshao** [comment]
I would suggest to make this configuration enabled by default, which is a convention for users to use Trino, and also aligned with Spark, what's your opinion @diqiu50 @yuqi1129 @xiacongling ?

**Comment by yuqi1129** [comment]
@jerryshao , I agree in general,  but should it include in release 0.5.0, I fear that many changes to tests need to be made.

**Comment by jerryshao** [comment]
let me think about this, but at least you should create an issue to track this.

**Pull request merged**